precompile/allow_list.go

```diff
@@ -70,6 +72,28 @@ func (c *AllowListConfig) Equal(other *AllowListConfig) bool {
 	return true
 }
 
+// Verify returns an error if there is an overlapping address between admins and enableds
+func (c *AllowListConfig) Verify() error {
+	// check if both lists are empty
```

Comment:
I think we can improve this by either:

1. Switch to returning early if one of the lists is empty since there can be no overlaps and move the more complex functionality out of the body of the conditional
2. This is cheap and only called once on startup, so we could also remove the check entirely

Reply:
I believe we do first one by checking `if len(c.EnabledAddresses) != 0 && len(c.AllowListAdmins) != 0 ` 
but yeah let's return early for readability